javalib/src/main/scala/java/util/concurrent/ConcurrentLinkedQueue.scala

```diff
@@ -0,0 +1,45 @@
+package java.util.concurrent
+
+import java.util._
+
+class ConcurrentLinkedQueue[E]()
+    extends AbstractQueue[E] with Queue[E] with Serializable {
+
+  def this(c: Collection[_ <: E]) = {
+    this()
+    addAll(c)
+  }
+
+  private val inner: LinkedList[E] = new LinkedList()
+
+  override def add(e: E): Boolean =
+    if (e == null) throw new NullPointerException()
+    else inner.add(e)
+
+  override def offer(e: E): Boolean =
+    add(e)
+
+  override def poll(): E =
+    inner.poll()
+
+  override def peek(): E =
+    inner.peek()
+
+  override def isEmpty(): Boolean =
+    inner.isEmpty()
+
+  override def size(): Int =
+    inner.size()
+
+  override def contains(o: Any): Boolean =
+    inner.contains(o)
+
+  override def remove(o: Any): Boolean =
+    inner.remove(o)
+
+  override def addAll(c: Collection[_ <: E]): Boolean =
+    inner.addAll(c)
+
+  override def iterator(): Iterator[E] =
+    inner.iterator()
```

Comment:
Why does this satisfy the weakly consistent property of `ConcurrentLinkedQueue`?

> Iterators are weakly consistent, returning elements reflecting the state of the queue at some point at or since the creation of the iterator. They do not throw ConcurrentModificationException, and may proceed concurrently with other operations. Elements contained in the queue since the creation of the iterator will be returned exactly once.

For example, how does it address the following test:
```scala
val queue = new ConcurrentLinkedQueue[Int]
queue.add(1)
queue.add(2)
val iter = queue.iterator()
expect(iter.next()).toEqual(1)
expect(iter.hasNext).toBeTruthy
queue.remove(2)
expect(iter.hasNext).toBeTruthy // yes, must still be truthy
expect(iter.next()).toEqual(2)
expect(iter.hasNext).toBeFalsy
```
Another one:
```scala
val queue = new ConcurrentLinkedQueue[Int]
queue.add(1)
queue.add(2)
queue.add(3)
val iter = queue.iterator()
expect(iter.next()).toEqual(1)
iter.remove(1)
iter.remove(3)
/* Since, we observed 1 before, and all the states of the queue since the creation of the
 * iterator that had 1 also had 2 and 3. So the iterator must also show 2 and 3.
 */
expect(iter.next()).toEqual(2)
expect(iter.next()).toEqual(3)
```
It is also unclear to me what should happen about calls to `remove()` on the iterator if they happen "concurrently" with modifications of the underlying queue. The Java spec doesn't seem to address this case, yet says that `remove()` is a supported operation.